src/app/shared/services/server-error/server-error.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, firstValueFrom, timer } from 'rxjs';
+import { TimerData, TimerFunctionReturn } from 'shared/models/server-error';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ServerErrorService {
```

Comment:
Use the service only to process requests to the api and set up a retry policy. Manage button status inside a component